include/envoy/thread_local/thread_local.h

```diff
@@ -76,11 +76,17 @@ class Slot {
   virtual void set(InitializeCb cb) PURE;
 
   /**
-   * UpdateCb takes the current stored data, and returns an updated/new version data.
-   * TLS will run the callback and replace the stored data with the returned value *in each thread*.
+   * UpdateCb takes the current stored data, and must return the same data. The
+   * API was designed to allow replacement of the object via this API, but this
+   * is not currently used, and thus we are removing the functionality. In a future
+   * PR, the API will be removed.
    *
-   * NOTE: The update callback is not supposed to capture the Slot, or its owner. As the owner may
-   * be destructed in main thread before the update_cb gets called in a worker thread.
+   * TLS will run the callback and assert the returned returned value matches
+   * the current value.
+   *
+   * NOTE: The update callback is not supposed to capture the Slot, or its
+   * owner. As the owner may be destructed in main thread before the update_cb
+   * gets called in a worker thread.
```

Comment:
nit: this is not a complete sentence; just replace '. As' with ', as'

Reply:
Good catch. That's preexisting though so I think I'll address in a follow-up PR since one's coming in this space anyway.